src/NHibernate/Transaction/AdoTransaction.cs

```diff
@@ -370,6 +370,7 @@ protected virtual void Dispose(bool isDisposing)
 					// don't dispose of multiple times.
 					return;
 				}
+				_isAlreadyDisposed = true;
```

Comment:
Change kept from #2144: flagging it at the end of this method instead of as soon as possible cause the method to be vulnerable to an endless loop, in case a `Dispose` is performed inside `AfterTransaction`, called later by this method.

This change was required by #2144 since it was calling `Dispose` in `AfterTransaction`. This PR here does not do it and so does not actually need that change, but still it is safer to have it.